sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -290,6 +290,7 @@ public GrowableOffsetRangeTracker restrictionTracker(
     return new GrowableOffsetRangeTracker(restriction.getFrom(), offsetPoller);
   }
 
+  @SuppressWarnings("PreferJavaTimeOverload")
```

Comment:
I dug into the test failure that you reported, the reason it works with `poll(long)` and not `poll(Duration)` is that we create our own MockConsumer, which only overrides `poll(long)`. If we just change this line to `poll(Duration)`: https://github.com/apache/beam/blob/414707077bb1f2689017d54076184c0b7d64955c/sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFnTest.java#L147

It will work for both cases. (Since MockConsumer.poll(long) defers to poll(Duration): https://github.com/apache/kafka/blob/2.4.1/clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java#L157-L159)